### PR TITLE
fix: Batch array length before sending its contents

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

**Comment by romange** [comment]
Ignore the failures, just want to see to see if it looks ok. I decided against a generic solution and instead chose whitelisting responses for  batching

**Review by dranikpg — APPROVED** [review]
Is it an experiment? Why did you choose only those two commands? I don't see much common in their reply structure (a few small indices vs probably string fields) 

I don't have a feel for it - never benchmarked different response types. But maybe we can still generalize it somehow, at least vaguely determine where it helps and where it makes it worse.

File: src/server/list_family.cc [hunk 1]
```diff
@@ -965,11 +965,12 @@ void ListFamily::LPos(CmdArgList args, ConnectionContext* cntx) {
       (*cntx)->SendLong((*result)[0]);
     }
   } else {
-    (*cntx)->StartArray(result->size());
+    (*cntx)->StartArray(result->size(), true);
     const auto& array = result.value();
     for (const auto& v : array) {
       (*cntx)->SendLong(v);
     }
+    (*cntx)->StopAggregate();
```
[the remark this answers is not in the document]
Comment (romange) [reply src/server/list_family.cc:973]: the output here is with a very high probability small and should be under a single packet hence I do not see any downside.

**Comment by romange** [comment]
> Is it an experiment?
> 
> > just want to see to see if it looks ok
> 
> So we'll add it manually? How will we determine for what commands?
> 
> I don't have a feel for it - never benchmarked different response types. But maybe we can still generalize it somehow, at least vaguely determine where it helps and where it makes it worse. I mean we must do it at the end to select the commands for which we enable those new checks (and only on certain conditions, etc)

I think that we won't add aggregation in general unless we see a problem like with MULTI. 
In fact, I am not even sure if we have other cases like this. Why?
the flow with multi is that we call `Send(...)` and then do a hop to run another program. Meanwhile, TCP sends the packet and during the next `Send` it's blocked on `ack`. 

for other cases, subsequent `Send` happen quickly enough so the TCP stack coalesces them together before sending out

**Review by dranikpg — APPROVED** [review]
🤷🏻‍♂️

**Comment by royjacobson** [comment]
I'm not a fan of the manual calls to `StopAggregate`. Could we use RAII for this?

```c++
struct SinkReplyBuilder {
  // ...
  struct ReplyAggregator {
    ReplyAggregator(SinkReplyBuilder* builder) : builder_(builder) {
      builder_->StartAggregate();
    }
    ~ReplyAggregator() {
      builder_->StopAggregate();
    }
    SinkReplyBuilder* builder_;
  };
  ReplyAggregator aggregate();
  // ...
};
```

**Comment by dranikpg** [comment]
True, but actually the simplest option is just to stop aggregating when the array is done (its used always like this in this PR)

**Comment by romange** [comment]
@dranikpg  I prefer not to do this because of the recursive nature of the reply builder. one can call StartArray multiple times.

**Review by dranikpg — APPROVED** [review]
About integer tracking - the depth can be tracked easily, so you change the number of elements only when you're on the same depth

**Pull request merged**